src/runtime-rs/src/shim/src/lib.rs

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) 2021 Alibaba Cloud
+// Copyright (c) 2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::os::unix::fs::FileTypeExt;
+use std::path::{Path, PathBuf};
+
+pub mod config;
+mod delete;
+mod event_publisher;
+mod run;
+mod start;
+mod task_service;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("failed to get bundle path")]
+    BundlePath(#[source] std::io::Error),
+    #[error("invalid argument")]
+    InvalidArgument,
+    #[error("failed to load oci spec")]
+    OciSpec(#[source] oci_spec::OciSpecError),
+    #[error("failed to extract oci spec information")]
+    OciSpecInfo(#[source] virtcontainers::oci::OciSpecInfoError),
+    #[error("failed to open file {1}")]
+    OpenFile(#[source] std::io::Error, std::path::PathBuf),
+    #[error("failed to read file {1}")]
+    ReadFile(#[source] std::io::Error, std::path::PathBuf),
+    #[error("failed to write file {1}")]
+    WriteFile(#[source] std::io::Error, std::path::PathBuf),
+    #[error("failed to get self exec")]
+    SelfExec(#[source] std::io::Error),
+    #[error("failed to bind socket at {1}")]
+    BindSocket(#[source] std::io::Error, String),
+    #[error("failed to spawn child")]
+    SpawnChild(#[source] std::io::Error),
+    #[error("failed to clean container {0}")]
+    CleanUp(String),
+    #[error("failed to get env variable")]
+    EnvVar(#[source] std::env::VarError),
+    #[error("failed to parse server fd {0}")]
+    ServerFd(String),
+    #[error("failed to start ttrpc server")]
+    StartServer(#[source] ttrpc::error::Error),
+    #[error("failed to wait ttrpc server when {0}")]
+    WaitServer(String),
+    #[error("failed to set logger")]
+    Logger,
+    #[error("error from config")]
+    Config(#[source] config::ConfigError),
+    #[error("error from sandbox")]
+    Sandbox(#[source] virtcontainers::Error),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+const SOCKET_ROOT: &str = "/run/containerd";
+
+const ENV_SOCKET_FD: &str = "BIND_FD";
```

Comment:
Is this a standardised name? Could it be changed to `KATA_BIND_FD`?

Reply:
Fixed in the first revision